tools/main.py

```diff
@@ -140,9 +144,13 @@ def main():
 
     if cfg.model.resume and check_isfile(cfg.model.resume):
         cfg.train.start_epoch = resume_from_checkpoint(
-            cfg.model.resume, model, optimizer=optimizer, scheduler=scheduler
-        )
+            cfg.model.resume, model, optimizer=optimizer,scheduler=scheduler
+            )
 
+    elif cfg.model.load_weights and check_isfile(cfg.model.load_weights):
+        load_pretrained_weights(model, cfg.model.load_weights)
+
```

Comment:
Kirill, is it possible to move these lines higher -- where `load_pretrained_weights` were earlier?
Is it possible to make loading pretrained weights or resuming _before_  `DataParallel` wrapping `model = DataParallel(model, ...)`

Why as about that:
* I think it is better load weights before DataParallel-related actions
* for NNCF wrapping it is more convenient

If from your point of view it is a wrong idea to move this block of code higher (may be due to some details concerning `resume`), I will make a workaround for NNCF, but, nevertheless, I think it is better to move it (if it is possible).

Reply:
done